pulsar/consumer_partition.go

```diff
@@ -479,7 +505,49 @@ func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, header
 		return err
 	}
 
-	uncompressedHeadersAndPayload, err := pc.Decompress(msgMeta, headersAndPayload)
+	decryptedPayload, err := pc.decryptor.Decrypt(headersAndPayload.ReadableSlice(), pbMsgID, msgMeta)
+	messages := make([]*message, 0)
+
+	// error decrypting the payload
+	if err != nil {
+		pc.log.Error(err)
+		switch pc.decryptor.CryptoFailureAction() {
+		case crypto.ConsumerCryptoFailureActionFail:
+			pc.log.Errorf("consuming message failed due to decryption err :%v", err)
```

Comment:
Let me take a look at this..

Reply:
I don't think we have this so may we just need to ack the message so it's not resent?

Reply:
yeah, but i don't think it is good idea to do `ack`.

Let's say user wants to consume this message again by providing proper crypto configuration, if we do `ack`, then he may not be able to consume.

Reply:
So, I think it's better to resend the message than doing `ack`. 
Any other thoughts here ??

Reply:
### java client
The purpose of java client adding msg id to `tracker` is to trigger redelivery of message request to broker, so that the message can be resent.

If it is not added to tracker it will not be resent to the  consumer until it restarted. And this tracker will be active only if `
[ackTimeoutMillis](https://pulsar.apache.org/docs/en/client-libraries-java/#configure-consumer)  config is provided.

## go client
> I don't think we have this so may we just need to ack the message so it's not resent?

If we leave the changes as it is this PR, i.e not adding msg id to tracker (since this client doesn't support this) => the message will not be resent.

[Acknowledgement timeout](https://pulsar.apache.org/docs/en/concepts-messaging/#acknowledgement-timeout)  this doc prefer negative acknowledgements over acknowledgement timeout. Java client is doing the later one(i.e triggering redelivery of message request on ack timeout).


So to be able to resent the message, we can do Nack (negative acknowledgement).

@cckellogg  what do you suggest ?

Reply:
good point maybe we should just Nack it here so it's redelivered on fail?

Reply:
Yes, we can do nack. I'll push changes with nack on failure.

Reply:
@cckellogg I've made the changes

Reply:
Looks good to me will you please fix the CI issue.

Reply:
thanks @cckellogg 

I don't think CI is failing due to last commit.

I see most of the open PR's are failing with the same issue as below.

`TestNamespaceTopicsNamespaceDoesNotExit (56.25s)
    client_impl_test.go:387: 
        	Error Trace:	client_impl_test.go:387
        	Error:      	Expected nil, but got: &errors.errorString{s:"server error: AuthorizationError: Exception occurred while trying to authorize GetTopicsOfNamespace"}
        	Test:       	TestNamespaceTopicsNamespaceDoesNotExit`

And last commit do not have any changes that effects `client_impl.go`. More likely this has something to do with broker config that CI uses for running test cases.

please let me know If I'm wrong here.

Reply:
@cckellogg looks like the CI is passed now. 
Please do suggest if there are any other improvements on this PR.